xdsl/xdsl_opt_main.py

```diff
@@ -26,7 +26,7 @@ class xDSLOptMain(CommandLineTool):
     stream.
     """
 
-    pipeline: list[ModulePass]
+    pipeline: PassPipelinePass
```

Comment:
While I agree that several passes can be viewed as one pass, I think the name `PassPipelinePass` will be quite confusing for people. The previous type annotation was more intuitive, I think.

Maybe we could rename this? Isn't this just essentially a class capturing a pass pipeline? So why not just `PassPipeline`?

Reply:
I agree, weirdly recurring, although I think `PipelinePass` might encapsulate our current naming convention better (A pipeline is a series of passes, but itself is also a pass).

Reply:
Yeah I like `PipelinePass` more than `PassPipeline` since I think it communicates more clearly that it's also a pass itself. I'll rename it.